Java/core/src/main/java/com/amazon/randomcutforest/tree/Node.java

```diff
@@ -422,4 +422,24 @@ protected void deleteSequenceIndex(long sequenceIndex) {
             sequenceIndexes.remove(sequenceIndex);
         }
     }
+
+    /**
+     * The following function returns the bounding box corresponding to the child
+     * that is not the path from the node to leaf traversed by a point. If such a
+     * box cannot be defined then the function returns null.
+     * 
+     * @param point the point as above
+     * @param node  the node as above
+     * @return the bounding box of the node
+     */
+
+    public static BoundingBox getSiblingBoundingBox(double[] point, Node node) {
+        if (node == null)
+            return null;
+        Node sibling = Node.isLeftOf(point, node) ? node.getRightChild() : node.getLeftChild();
+        if (sibling == null)
+            return null;
```

Comment:
It's only the root node that doesn't have a sibling, right? We could have this method return an `Optional<BoundingBox>` instead, although that would create a transient wrapper class which I know you're trying to avoid.

Reply:
The "sibling" is not the sibling of the node, it is the sibling of the path taken by point. Will change the name, it may avoid confusion.

More specifically, the function should return null for any leaf node (because the path is non-existent).